Userland/Libraries/LibGfx/ImageFormats/JPEGWriter.cpp

```diff
@@ -359,35 +367,55 @@ class JPEGEncodingContext {
             auto coefficient = component[zigzag_map[k]];
             if (coefficient == 0) {
                 if (k == 63) {
-                    TRY(write_symbol(ac_table.from_input_byte(0x00)));
+                    TRY(m_symbols_and_bits.try_append(Symbol { .byte = 0x00, .component_id = component_id, .is_dc = false }));
                     break;
                 }
                 r += 1;
                 continue;
             }
 
             while (r > 15) {
-                TRY(write_symbol(ac_table.from_input_byte(0xF0)));
+                TRY(m_symbols_and_bits.try_append(Symbol { .byte = 0xF0, .component_id = component_id, .is_dc = false }));
                 r -= 16;
             }
 
             {
                 // F.3 - Sequential encoding of a non-zero AC coefficient
                 auto const ssss = csize(coefficient);
-                auto const rs = (r << 4) + ssss;
-                TRY(write_symbol(ac_table.from_input_byte(rs)));
+                u8 const rs = (r << 4) + ssss;
+                TRY(m_symbols_and_bits.try_append(Symbol { .byte = rs, .component_id = component_id, .is_dc = false }));
 
                 if (coefficient < 0)
                     coefficient -= 1;
 
-                TRY(m_bit_stream.write_bits<u16>(coefficient, ssss));
+                TRY(m_symbols_and_bits.try_append(RawBits(coefficient, ssss)));
             }
 
             r = 0;
         }
         return {};
     }
 
+    ErrorOr<void> write_symbols_to_stream()
+    {
+        for (auto const& symbol_or_bits : m_symbols_and_bits) {
+            if (symbol_or_bits.has<Symbol>()) {
```

Comment:
nit: If you go with Variant, I like using `visit()`, which statically makes sure you don't forget to handle any cases. (Here, there are just two, and there won't be more, so I suppose it's fine as-is.)

Reply:
> I like using visit(), which statically makes sure you don't forget to handle any cases.

I should keep that in mind more.